### ISSUE TITLE
[question] Confusion regarding default tool title and description.

### ISSUE THREAD

**Issue by AdamRobertson-Cognasim** [BrainBlend-AI/atomic-agents#47]
I'm enjoying working with your framework so far! It is very helpful.

Small question however: Why is the default title and descriptions for tools based on the input schema? Asking because I want to make sure I'm understanding the motivation so I can take advantage of it.

Is this so that a tool can have multiple uses by changing what input_schema is being used in config?
If that's so, then isn't it less confusing to just change the description from config? And have the tools run function accept a Union of all acceptable schemas?

Having it default to the input schemas caught me by surprise and seems unintuitive.

**Comment by KennyVaneetvelde** [comment]
@AdamRobertson-Cognasim 

You are correct in your assumption of its usage, but incorrect in your assumption of how it works 😅 But to be fair this is probably the only thing in the framework I didn't obsessively over-think... Mostly because I couldn't envision many clear use cases, so I'm open to ideas and scenarios on how you want to be using this!

Anyways, if you look closely at the code of `base_tool.py` you will see that in fact you can change the description from the config, and it is in fact the default. The title & description in the input schema are a fallback. Looking at it know, maybe it makes sense to only have it in the config...

But for now, I'd say use whatever way makes most sense to you and stick by it...

For now, I won't change anything but leave this issue open in case anyone wants to join in with some input

**Comment by AdamRobertson-Cognasim** [comment]
Ok! Just curious.

**Comment by KennyVaneetvelde** [comment]
Closing issue due to being stale and naming has changed in v2.0 either way
